File: deltafi-core-domain/src/main/java/org/deltafi/core/domain/api/types/DeltaFiles.java
/**
 *    DeltaFi - Data transformation and enrichment platform
 *
 *    Copyright 2022 DeltaFi Contributors <deltafi@deltafi.org>
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */
package org.deltafi.core.domain.api.types;

import java.util.List;

/*
 * This is the codegen generated class, except the generated DeltaFile is replaced with org.deltafi.core.domain.api.types.DeltaFile
 */
public class DeltaFiles {
  private Integer offset;

  private Integer count;

  private Integer totalCount;

  private List<DeltaFile> deltaFiles;

  public DeltaFiles() {
  }

  @SuppressWarnings("unused")
  public DeltaFiles(Integer offset, Integer count, Integer totalCount, List<DeltaFile> deltaFiles) {
    this.offset = offset;
    this.count = count;
    this.totalCount = totalCount;
    this.deltaFiles = deltaFiles;
  }

  public Integer getOffset() {
    return offset;
  }

  public void setOffset(Integer offset) {
    this.offset = offset;
  }

  public Integer getCount() {
    return count;
  }

  public void setCount(Integer count) {
    this.count = count;
  }

  public Integer getTotalCount() {
    return totalCount;
  }

  public void setTotalCount(Integer totalCount) {
    this.totalCount = totalCount;
  }

  public List<DeltaFile> getDeltaFiles() {
    return deltaFiles;
  }

  public void setDeltaFiles(List<DeltaFile> deltaFiles) {
    this.deltaFiles = deltaFiles;
  }

  @Override
  public String toString() {
    return "DeltaFiles{" + "offset='" + offset + "'," +"count='" + count + "'," +"totalCount='" + totalCount + "'," +"deltaFiles='" + deltaFiles + "'" +"}";
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        DeltaFiles that = (DeltaFiles) o;
        return java.util.Objects.equals(offset, that.offset) &&
                            java.util.Objects.equals(count, that.count) &&
                            java.util.Objects.equals(totalCount, that.totalCount) &&
                            java.util.Objects.equals(deltaFiles, that.deltaFiles);
  }

  @Override
  public int hashCode() {
    return java.util.Objects.hash(offset, count, totalCount, deltaFiles);
  }

  public static Builder newBuilder() {
    return new Builder();
  }

  public static class Builder {
    private Integer offset;

    private Integer count;

    private Integer totalCount;

    private List<DeltaFile> deltaFiles;

    public DeltaFiles build() {
                  DeltaFiles result = new DeltaFiles();
                      result.offset = this.offset;
          result.count = this.count;
          result.totalCount = this.totalCount;
          result.deltaFiles = this.deltaFiles;
                      return result;
    }

    public Builder offset(Integer offset) {
      this.offset = offset;
      return this;
    }

    public Builder count(Integer count) {
      this.count = count;
      return this;
    }

    public Builder totalCount(
        Integer totalCount) {
      this.totalCount = totalCount;
      return this;
    }

    public Builder deltaFiles(
        List<DeltaFile> deltaFiles) {
      this.deltaFiles = deltaFiles;
      return this;
    }
  }
}

File: tui/cmd/deltafiles.go
/*
 *    DeltaFi - Data transformation and enrichment platform
 *
 *    Copyright 2021-2025 DeltaFi Contributors <deltafi@deltafi.org>
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */
package cmd

import (
	"fmt"

	"github.com/spf13/cobra"
)

var deltafilesCmd = &cobra.Command{
	Use:   "deltafiles",
	Short: "Manage and search DeltaFiles",
	Long: `Manage and search DeltaFiles in the DeltaFi system.

This command provides tools for searching, analyzing, and managing DeltaFiles
in the system. You can search for specific patterns, view summaries, and
perform bulk operations on files.

Examples:
  deltafi deltafiles view                    # Interactive DeltaFiles viewer
  deltafi deltafiles list                    # List DeltaFiles`,
	GroupID:            "deltafile",
	SilenceUsage:       true,
	DisableSuggestions: true,
	RunE: func(cmd *cobra.Command, args []string) error {
		cmd.Help()
		return fmt.Errorf("subcommand is required")
	},
}

var deltafilesViewCmd = &cobra.Command{
	Use:          "view",
	Short:        "View DeltaFiles",
	Long:         `View DeltaFiles using various criteria such as flow names, time ranges, or other filters`,
	SilenceUsage: true,
	RunE:         runDeltafilesView,
}

func runDeltafilesView(cmd *cobra.Command, args []string) error {
	RequireRunningDeltaFi()
	if len(args) > 0 {
		return fmt.Errorf("deltafiles view command does not accept positional arguments: %v", args)
	}

	filter, err := searchParams.GetFilter()
	if err != nil {
		return err
	}

	return runSearchViewer(filter)
}

var deltafilesListCmd = &cobra.Command{
	Use:          "list",
	Short:        "List DeltaFiles",
	Long:         `List DeltaFiles in a simple format.`,
	SilenceUsage: true,
	RunE:         runDeltafilesList,
}

func runDeltafilesList(cmd *cobra.Command, args []string) error {
	RequireRunningDeltaFi()
	if len(args) > 0 {
		return fmt.Errorf("list command does not accept positional arguments: %v", args)
	}

	plain, _ := cmd.Flags().GetBool("plain")
	json, _ := cmd.Flags().GetBool("json")
	yaml, _ := cmd.Flags().GetBool("yaml")
	limit, _ := cmd.Flags().GetInt("limit")
	offset, _ := cmd.Flags().GetInt("offset")
	verbose, _ := cmd.Flags().GetBool("verbose")
	terse, _ := cmd.Flags().GetBool("terse")

	filter, err := searchParams.GetFilter()
	if err != nil {
		return err
	}

	if json {
		return searchResultJSON(filter, limit, offset, plain, verbose)
	} else if yaml {
		return searchResultYAML(filter, limit, offset, verbose)
	} else {
		return searchResultTable(filter, limit, offset, plain, terse)
	}
}

func init() {
	rootCmd.AddCommand(deltafilesCmd)
	deltafilesCmd.AddCommand(deltafilesViewCmd)
	deltafilesCmd.AddCommand(deltafilesListCmd)
	searchParams.addSearchFlagsAndCompletions(deltafilesViewCmd)

	searchParams.addSearchFlagsAndCompletions(deltafilesListCmd)
	deltafilesListCmd.Flags().BoolP("plain", "p", false, "Show plain table output")
	deltafilesListCmd.Flags().Bool("json", false, "Show JSON output")
	deltafilesListCmd.Flags().Bool("yaml", false, "Show YAML output")
	deltafilesListCmd.Flags().Int("limit", 100, "Limit the number of results")
	deltafilesListCmd.Flags().Int("offset", 0, "Offset the results")
	deltafilesListCmd.Flags().BoolP("verbose", "v", false, "Show verbose output (JSON and YAML only)")
	deltafilesListCmd.Flags().Bool("terse", false, "Hide table header and footer")
}
